### PR TITLE
[FlagScale] Split args associated with process_mesh

### PR DESCRIPTION
**Pull request by heavyrain-lzy**
Update the `args` and `xxx_config` according to  the associated process_mesh.

### CODE REVIEW

File: flagscale/train/arguments.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            return self._rank_mapper
+        return getattr(self.args, name)
+
+    def __str__(self):
+        return str(self.args)
+
+    def __repr__(self):
+        return repr(self.args)
+    
+    def _initialize_distributed(self):
```
Comment (aoyulong) [inline flagscale/train/arguments.py:27]: Why do we need to move this distributed initialization here?

File: flagscale/train/arguments.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    def _build_rank_mapper(self):
+        self._initialize_distributed()
+        self._rank_mapper = RankMapper(self.args)
+        return self._rank_mapper
+
+    @classmethod
+    def from_dict(cls, args):
+        return cls(args)
+
+    def pre_validate_args(self):
```
Comment (aoyulong) [inline flagscale/train/arguments.py:69]: The arguments extended by flagscale were split into two parts for now. However, in the future, the arguments may have more than two parts.

File: flagscale/train/arguments.py (hunk 1 above)
Comment (heavyrain-lzy) [reply flagscale/train/arguments.py:27]: To create the map between `logical_rank` and `physical rank`.

File: flagscale/train/arguments.py (hunk 2 above)
Comment (heavyrain-lzy) [reply flagscale/train/arguments.py:69]: To decrease the code change in the `Megatron-LM` repo, limit modifications to the `validate_args` function in the original argument file when adding new parameters.

File: megatron/megatron/training/initialize.py [hunk 3]
```diff
@@ -274,7 +282,7 @@ def _initialize_distributed(get_embedding_ranks, get_position_embedding_ranks):
     # data-parallel communicators.
     if device_count > 0:
         # Set the parallel context.
-        if args.enable_hetero:
```
Comment (aoyulong) [inline megatron/megatron/training/initialize.py]: So we can remove the `args.enbale_hetero` from the code and use `args.hetero_process_meshes` instead?

**Review by aoyulong — COMMENTED** [review]
We should design a better mechanism to extend new args by FlagScale.

File: megatron/megatron/training/initialize.py (hunk 3 above)
Comment (heavyrain-lzy) [reply megatron/megatron/training/initialize.py]: I think that the way using `process_meshes` to configure the training plan is not only designed for `heterogeneous training`, but also for other multi-model training like `Llava`. So we should unbind the `args.enbale_hetero` parameter and the `hetero_process_meshes` parameter gradually.

**Review by aoyulong — APPROVED** [review]
LGTM

**Pull request merged**